jsonschema/tests/test_validators.py

```diff
@@ -1186,14 +1186,21 @@ def test_it_retrieves_unstored_refs_via_requests(self):
     def test_it_retrieves_unstored_refs_via_urlopen(self):
         ref = "http://bar#baz"
         schema = {"baz": 12}
+        resource_manager_mock = mock.MagicMock()
+        (resource_manager_mock.
+         __enter__.return_value.
+         read.return_value.
+         decode.return_value) = json.dumps(schema).encode("utf8")
```

Comment:
This doesn't look correct, and is a good example of why mocking is sadness :/

`decode` returns unicode, not bytes.

Will fix on merge though, thanks.